glean-core/src/metrics/mod.rs

```diff
@@ -28,6 +29,10 @@ use crate::Glean;
 pub use self::boolean::BooleanMetric;
 pub use self::counter::CounterMetric;
 pub use self::datetime::DatetimeMetric;
+pub(crate) use self::experiment::ExperimentMetric;
+// Note: clippy complains about this being useless but, if I remove
+// the next line, then tests in glean-core\src\lib.rs will fail to build.
+pub(crate) use self::experiment::RecordedExperimentData;
```

Comment:
Two possibilities:

1. Only export it in test mode:

```suggestion
#[cfg(test)]
pub(crate) use self::experiment::RecordedExperimentData;
```

2. Below in the `pub enum Metric` just use `RecordedExperimentData`, not `experiment::RecordedExperimentData`.

Both should make the clippy complain go away

Reply:
Thanks for this. I had to go for the (1), the (2) would make the build fail.